### PR TITLE
[BugFix] BE crash when base rowset is missing

### LINKED ISSUE #10008: BE crash when compaction
### Steps to reproduce the behavior (Required)
BE crash when running primary stability cases:

```
Core was generated by `/home/disk2/sr/stability_master_01/be-50b527ce-645d-4e6a-bfc6-1e5e231d7cff/lib/'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count (__r=..., Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0x7f71280f3078)
    at /root/starrocks/be/src/storage/base_and_cumulative_compaction_policy.cpp:161
161	/root/starrocks/be/src/storage/base_and_cumulative_compaction_policy.cpp: No such file or directory.
[Current thread is 1 (Thread 0x7f71280f9700 (LWP 65131))]
(gdb) bt
Python Exception <type 'exceptions.ImportError'> No module named gdb.frames:
#0  std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count (__r=..., this=0x7f71280f3078)
    at /root/starrocks/be/src/storage/base_and_cumulative_compaction_policy.cpp:161
#1  std::__shared_ptr<starrocks::Rowset, (__gnu_cxx::_Lock_policy)2>::__shared_ptr<starrocks::Rowset, void> (__r=..., this=0x7f71280f3070)
    at /usr/include/c++/10.3.0/bits/shared_ptr_base.h:1207
#2  std::shared_ptr<starrocks::Rowset>::shared_ptr<starrocks::Rowset, void> (__r=..., this=0x7f71280f3070) at /usr/include/c++/10.3.0/bits/shared_ptr.h:325
#3  std::enable_shared_from_this<starrocks::Rowset>::shared_from_this (this=0x8) at /usr/include/c++/10.3.0/bits/shared_ptr.h:807
#4  starrocks::BaseAndCumulativeCompactionPolicy::_pick_base_rowsets (this=0x246d1b090, rowsets=0x7f71280f30f0)
    at /root/starrocks/be/src/storage/base_and_cumulative_compaction_policy.cpp:161
#5  0x0000000001eeaebb in starrocks::BaseAndCumulativeCompactionPolicy::_create_base_compaction (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0x246d1b090)
    at /root/starrocks/be/src/storage/base_and_cumulative_compaction_policy.cpp:184
#6  0x0000000001eec6fb in starrocks::BaseAndCumulativeCompactionPolicy::create_compaction (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0x246d1b090)
    at /root/starrocks/be/src/storage/base_and_cumulative_compaction_policy.cpp:34
#7  0x0000000001d3eb98 in starrocks::Tablet::get_compaction (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0x18b93c00, type=<optimized out>, create_if_not_exist=create_if_not_exist@entry=true)
    at /root/starrocks/be/src/storage/tablet.cpp:1297
#8  0x0000000001ee74e2 in starrocks::CompactionScheduler::_can_do_compaction (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
candidate=..., need_reschedule=0x7f71280f331f, compaction_task=0x7f71280f3320)
    at /root/starrocks/be/src/storage/compaction_scheduler.cpp:182
#9  0x0000000001ee8305 in starrocks::CompactionScheduler::_try_get_next_compaction_task (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=<optimized out>) at /root/starrocks/be/src/storage/compaction_scheduler.cpp:228
#10 0x0000000001ee8a29 in starrocks::CompactionScheduler::schedule (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=this@entry=0x7f71280f3630) at /root/starrocks/be/src/storage/compaction_scheduler.cpp:36
#11 0x0000000001f13f7e in operator() (__closure=<optimized out>) at /root/starrocks/be/src/storage/olap_server.cpp:122
#12 std::__invoke_impl<void, starrocks::StorageEngine::start_bg_threads()::<lambda()> > (__f=...) at /usr/include/c++/10.3.0/bits/invoke.h:60
#13 std::__invoke<starrocks::StorageEngine::start_bg_threads()::<lambda()> > (__fn=...) at /usr/include/c++/10.3.0/bits/invoke.h:95
#14 std::thread::_Invoker<std::tuple<starrocks::StorageEngine::start_bg_threads()::<lambda()> > >::_M_invoke<0> (this=<optimized out>) at /usr/include/c++/10.3.0/thread:264
#15 std::thread::_Invoker<std::tuple<starrocks::StorageEngine::start_bg_threads()::<lambda()> > >::operator() (this=<optimized out>) at /usr/include/c++/10.3.0/thread:271
#16 std::thread::_State_impl<std::thread::_Invoker<std::tuple<starrocks::StorageEngine::start_bg_threads()::<lambda()> > > >::_M_run(void) (this=<optimized out>)
    at /usr/include/c++/10.3.0/thread:215
#17 0x000000000607a320 in std::execute_native_thread_routine (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
__p=0xef46530) at ../../../.././libstdc++-v3/src/c++11/thread.cc:80
#18 0x00007f722ca79ea5 in start_thread () from /lib64/libpthread.so.0
Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
#19 0x00007f722c094b0d in clone () from /lib64/libc.so.6
(gdb)
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+--------------------+
| current_version()  |
+--------------------+
| 2.4.0-0815 f5c2bd7 |
+--------------------+
1 row in set (0.06 sec)
```

### PR DESCRIPTION
**Pull request by rickif**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #10008

## Problem Summary(Required) ：

The base rowset (rowset whose start version is 0) may be missing for some tablets. This PR check the base rowset to avoid the crash in compaction.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by rickif** [comment]
run starrocks_be_unittest

**Review by chaoyli — DISMISSED** [review]

File: be/src/storage/base_and_cumulative_compaction_policy.cpp [hunk 1]
```diff
@@ -158,6 +158,9 @@ void BaseAndCumulativeCompactionPolicy::_pick_base_rowsets(std::vector<RowsetSha
     size_t rowsets_compaction_score = 0;
     // add the base rowset to input_rowsets
     Rowset* base_rowset = *_compaction_context->rowset_levels[2].begin();
+    if (base_rowset == nullptr) {
```
Comment (ABingHuang) [inline be/src/storage/base_and_cumulative_compaction_policy.cpp:161]: in which situation there is no base rowset?

Comment (rickif) [reply be/src/storage/base_and_cumulative_compaction_policy.cpp:161]: The clone of tablet may lead no base rowset.

**Review by meegoo — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by rickif** [comment]
@mergifyio rebase

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by rickif** [comment]
@mergifyio backport branch-2.4

**Pull request merged**